### DISCUSSION TITLE
multi-select не работает с фильтром

### DISCUSSION THREAD

**Discussion by expdts** [taiga-family/taiga-ui#7356] [Q&A] [answered]
Подскажите пожалуйста, почему в моем случае не работает выпадающий список?
[Пример](https://stackblitz.com/edit/angular-tmuv65-gklrq1?file=src%2Fapp%2Fapp.component.html). При открытии списка, список пустой, если выполнить поиск, то данные появляются.

**Comment by expdts** [comment]
@waterplea причину ошибки нашел. Если данные для выпадающего списка передать через `ngOnInit`, а не задать изначально, то список для выбора не будет отображаться (пример 1).
Также перестал работать поиск (пример 2), заметил только сейчас, возможно 1й пример поэтому же не работает. До этого поиск в `multi-select` работал в версии `3.67.0`, сейчас использую `3.76.0`

**Reply by waterplea** [reply]
Изначально ничего не выводится, потому что ничего не стартует поток — можно переделать на `searchTestListData$ = new BehaviorSubject('')` чтобы стартовало с пустой строки. Во втором примере не работает поиск, потому что он стартует с `requestListData$`, а в него ничего не некстит при вводе.

**Reply by expdts** [reply]
> BehaviorSubject('')

Спасибо, это сработало.
Но после поиска и/или выбора, в выпадающем списке не отображаются остальные данные, видео прилагаю. Можете подсказать, как это исправить?

https://github.com/taiga-family/taiga-ui/assets/109337333/0d3501ed-982d-4cc4-bcf3-0cabeb02bbc1

**Reply by waterplea** [reply, marked as answer]
Когда ты выбрал элемент и это автоматом стёрло значение — в searchChange летит null, чтобы можно было отличить от пользователя, который стёр весь текст. Тебе, наверное, нужно его обрабатывать как пустую строку.
